File: checkmate-spring-kafka/src/main/java/io/github/malczuuu/checkmate/spring/kafka/LocalUtils.java
package io.github.malczuuu.checkmate.spring.kafka;

class LocalUtils {

  /**
   * Checks whether the Kafka client library is present on the classpath by attempting to load a
   * core class.
   *
   * @return {@code true} if the class is found, or {@code false} if a {@code
   *     ClassNotFoundException} is thrown
   */
  static boolean isKafkaPresent() {
    try {
      Class.forName("org.apache.kafka.clients.consumer.KafkaConsumer");
      return true;
    } catch (ClassNotFoundException e) {
      return false;
    }
  }
}

File: checkmate-spring-kafka/src/main/kotlin/io/github/malczuuu/checkmate/spring/kafka/LocalUtils.kt
package io.github.malczuuu.checkmate.spring.kafka

/** Utility methods for local use within this module. */
internal object LocalUtils {

  /**
   * Checks whether the Kafka client library is present on the classpath by attempting to load a
   * core class.
   *
   * @returns `true` if the class is found, or `false` if a `ClassNotFoundException` is thrown
   */
  fun isKafkaPresent(): Boolean {
    try {
      Class.forName("org.apache.kafka.clients.consumer.KafkaConsumer")
      return true
    } catch (_: ClassNotFoundException) {
      return false
    }
  }
}
